pandas/core/frame.py

```diff
@@ -5033,10 +5036,10 @@ def drop(  # type: ignore[override]
     @overload
     def rename(
         self,
-        mapper: Renamer | None = ...,
+        mapper: Renamer[HashableT] | None = ...,
         *,
-        index: Renamer | None = ...,
-        columns: Renamer | None = ...,
+        index: Renamer[HashableTa] | None = ...,
+        columns: Renamer[HashableTb] | None = ...,
```

Comment:
ugly: need different TypeVars so that they are not tied to each other

Reply:
maybe add the comment here as well